File: backend/src/main/java/com/project2/domain/post/entity/Comment.java
package com.project2.domain.post.entity;

import com.project2.domain.member.entity.Member;
import com.project2.global.entity.BaseTime;

import com.project2.global.exception.ServiceException;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Entity
public class Comment extends BaseTime {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "comment_id")
	private Long id;

	@Column(nullable = false, columnDefinition = "TEXT")
	private String content;

	private int depth;

	@ManyToOne
	@JoinColumn(nullable = false)
	private Post post;

	@ManyToOne
	@JoinColumn(nullable = false)
	private Member member;

	@ManyToOne
	@JoinColumn(name = "parent_comment_id")
	private Comment parent;

	@OneToMany(mappedBy = "parent", cascade = CascadeType.REMOVE, orphanRemoval = true)
	@Builder.Default
	private List<Comment> children = new ArrayList<>();

	public void updateContent(String newContent) {
		if (newContent == null || newContent.trim().isEmpty()) {
			throw new ServiceException("400", "댓글 내용은 비어 있을 수 없습니다.");
		}
		this.content = newContent;
	}
}

File: backend/src/main/kotlin/com/project2/domain/post/entity/Comment.kt
package com.project2.domain.post.entity

import com.project2.domain.member.entity.Member
import com.project2.global.entity.BaseTime
import com.project2.global.exception.ServiceException
import jakarta.persistence.*

@Entity
class Comment : BaseTime() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comment_id")
    var id: Long? = null

    @Column(nullable = false, columnDefinition = "TEXT")
    lateinit var content: String

    var depth: Int = 0

    @ManyToOne
    @JoinColumn(nullable = false)
    lateinit var post: Post

    @ManyToOne
    @JoinColumn(nullable = false)
    lateinit var member: Member

    @ManyToOne
    @JoinColumn(name = "parent_comment_id")
    var parent: Comment? = null

    @OneToMany(mappedBy = "parent", cascade = [CascadeType.REMOVE], orphanRemoval = true)
    var children: MutableList<Comment> = mutableListOf()

    fun updateContent(newContent: String) {
        if (newContent.isBlank()) {
            throw ServiceException("400", "댓글 내용은 비어 있을 수 없습니다.")
        }
        this.content = newContent
    }
}